### ISSUE TITLE
Animation resource showing up multiple times in editor

### ISSUE THREAD

**Issue by plojo** [JirkaDellOro/FUDGE#308]
When attaching an animator component to any node said animator creates an empty animation (resource):

![image](https://user-images.githubusercontent.com/38690628/164673937-94b050f9-9d72-4b63-820c-fc6823dd62df.png)

After saving the project Internal.json contains said animator and its corresponding animation:

![image](https://user-images.githubusercontent.com/38690628/164674913-e1522e55-002b-4f8e-99f9-512ab6d9721c.png)

If the project is then reloaded again multiple animation resources start showing up:

![image](https://user-images.githubusercontent.com/38690628/164675602-ff09fbb6-bfec-4aae-9a12-4341917661ec.png)

saving and reloading repeatedly leads to more and more animation resources showing up in editor as well as in Internal.json:

![image](https://user-images.githubusercontent.com/38690628/164676200-9728d159-2b75-4a4a-a471-82c8b70d84d2.png)

**Comment by JirkaDellOro** [comment]
Please test. Make sure, that the ids of the animations match especially the key and the idResource-field. There were inconsistencies due to an older loading algorithm

**Comment by plojo** [comment]
Ok, animations still seem to multiply but more slowly, but I found out why:
While the _ComponentAnimator_ (not the _Animation_) gets reconstructed (during deserialization of the _Graph_) a new _ComponentAnimator_ is created with an empty constructor call. Calling _ComponentAnimator_ constructor without parameters leads to the creation of a new _Animation_ which then registers it self to the project. So each time the _Graph_  gets deserialized we ended up creating an additional new Animation.
